### PR TITLE
[Merged by Bors] - chore: Rename more `coe_nat`/`coe_int` to `natCast`/`intCast`

### PR DESCRIPTION
**Pull request by YaelDillies**
---

- [x] depends on: #13197

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Int/GCD.lean [hunk 1]
```diff
@@ -169,9 +169,11 @@ namespace Int
 
 theorem gcd_def (i j : ℤ) : gcd i j = Nat.gcd i.natAbs j.natAbs := rfl
 
-protected theorem coe_nat_gcd (m n : ℕ) : Int.gcd ↑m ↑n = Nat.gcd m n :=
+protected theorem natCast_gcd (m n : ℕ) : Nat.gcd m n = Int.gcd ↑m ↑n :=
   rfl
-#align int.coe_nat_gcd Int.coe_nat_gcd
+#align int.coe_nat_gcd Int.natCast_gcd
+
+@[deprecated (since := "2024-05-25")] alias coe_nat_gcd := Int.natCast_gcd
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Int/GCD.lean:176]: Please keep the original direction for the deprecated version

**Review by Ruben-VandeVelde — APPROVED** [review]

File: Mathlib/Data/Int/GCD.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Data/Int/GCD.lean]: I've split that change to #13221

Comment (YaelDillies) [reply Mathlib/Data/Int/GCD.lean]: Oh, I see what you mean. I had assumed `Int.gcd` what `Int`-valued

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Data/Int/Cast/Lemmas.lean [hunk 2]
```diff
@@ -205,9 +205,11 @@ theorem cast_natAbs : (n.natAbs : α) = |n| := by
 
 end LinearOrderedRing
 
-theorem coe_int_dvd [CommRing α] (m n : ℤ) (h : m ∣ n) : (m : α) ∣ (n : α) :=
+theorem cast_dvd_cast [CommRing α] (m n : ℤ) (h : m ∣ n) : (m : α) ∣ (n : α) :=
```
Comment (sgouezel) [inline Mathlib/Data/Int/Cast/Lemmas.lean:208]:
```suggestion
theorem intCast_dvd_intCast [CommRing α] (m n : ℤ) (h : m ∣ n) : (m : α) ∣ (n : α) :=
```
for coherence?

File: Mathlib/Algebra/Order/Group/Int.lean [hunk 3]
```diff
@@ -34,8 +34,10 @@ open Function Nat
 
 namespace Int
 
-theorem coe_nat_strictMono : StrictMono (· : ℕ → ℤ) := fun _ _ ↦ Int.ofNat_lt.2
-#align int.coe_nat_strict_mono Int.coe_nat_strictMono
+theorem natCast_strictMono : StrictMono (· : ℕ → ℤ) := fun _ _ ↦ Int.ofNat_lt.2
```
Comment (sgouezel) [inline Mathlib/Algebra/Order/Group/Int.lean:37]: Does the naming convention specify the order of words here, i.e., `natCast_strictMono` or `strictMono_natCast` (where the latter follows more closely the order of symbols in the statement)? Greping mathlib, I found several examples in both directions.

File: Mathlib/Data/Int/Cast/Lemmas.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/Data/Int/Cast/Lemmas.lean:208]: It's in namespace `Int` unless I've messed up.

File: Mathlib/Algebra/Order/Group/Int.lean (hunk 3 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Order/Group/Int.lean:37]: `injective`, `mono`, `strictMono` go to the end. Most other predicates go to the front.

File: Mathlib/Data/Int/Cast/Lemmas.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/Data/Int/Cast/Lemmas.lean:208]: Indeed, I just checked and it is in namespace `Int`

File: Mathlib/Algebra/Order/Group/Int.lean (hunk 3 above)
Comment (sgouezel) [reply Mathlib/Algebra/Order/Group/Int.lean:37]: I could not find comments about this in https://leanprover-community.github.io/contribute/naming.html. Could you maybe update this page (and try to explain the convention, which looks pretty arbitrary to me)?

Comment (YaelDillies) [reply Mathlib/Algebra/Order/Group/Int.lean:37]: I will do that, but can we not hold up this PR on it? It's completely orthogonal.

**Comment by erdOne** [comment]
I think CI failed at `Mathlib.Order.Grade.lean` on this PR.

**Comment by sgouezel** [comment]
bors r-
bors d+

**Comment by YaelDillies** [comment]
bors merge

**Comment by YaelDillies** [comment]
Let's try again.

bors merge

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**